aten/src/ATen/native/BinaryOps.h

```diff
@@ -9,7 +9,33 @@ struct TensorIterator;
 struct TensorIteratorBase;
 }
 
-namespace at { namespace native {
+namespace at {
+namespace native {
+at::Tensor& add_out(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    const Scalar& alpha,
+    Tensor& out);
+
+at::Tensor& add_out(
+    Tensor& out,
+    const at::Tensor& self,
+    const at::Tensor& other,
+    const Scalar& alpha = 1);
+
+at::Tensor wrapper_add_Tensor(
```

Comment:
nit (no need to make any changes): you don't need to bother exposing these wrappers in the header, since they're just an implementation detail of `at::native::add()`. And then your other API's (`at::add` and at::Tensor::add`) just call into the native function.